packages/content/src/content.ts

```diff
@@ -19,7 +19,12 @@ export function getContent(
   if (requestOptions && requestOptions.isPortal) {
     return getContentFromPortal(id, requestOptions);
   } else {
-    return getContentFromHub(id, requestOptions);
+    return getContentFromHub(id, requestOptions).catch(() => {
+      // TODO: inspect error?
+      // dataset is not in index (i.e. might be a private item)
+      // try fetching from portal instead
+      return getContentFromPortal(id, requestOptions);
```

Comment:
is this supported for all item types? if somebody is selecting a feature layer dataset, a failed request would return a feature service item. is that acceptable?

Reply:
good point, but I think that is out of scope of this PR, b/c `getContent()` doesn't yet support feature layer/service datasets.

We should address this as part of #322